test/functional/page_objects/settings_page.ts

```diff
@@ -486,6 +486,55 @@ export function SettingsPageProvider({ getService, getPageObjects }: FtrProvider
       await this.clickSaveScriptedField();
     }
 
+    async addRuntimeField(name: string, type: string, script: string) {
+      await this.clickAddField();
+      await this.setFieldName(name);
+      await this.setFieldType(type);
+      if (script) {
+        await this.setFieldScript(script);
+      }
+      await this.clickSaveField();
+      await this.closeIndexPatternFieldEditor();
+    }
+
+    async closeIndexPatternFieldEditor() {
+      await retry.waitFor('field editor flyout to close', async () => {
+        return !(await testSubjects.exists('euiFlyoutCloseButton'));
+      });
+    }
+
+    async clickAddField() {
+      log.debug('click Add Field');
+      await testSubjects.click('addField');
+    }
+
+    async clickSaveField() {
+      log.debug('click Save');
+      await testSubjects.click('fieldSaveButton');
+    }
+
+    async setFieldName(name: string) {
+      log.debug('set field name = ' + name);
+      await testSubjects.setValue('nameField', name);
+    }
+
+    async setFieldType(type: string) {
+      log.debug('set type = ' + type);
+      await testSubjects.setValue('typeField', type);
+    }
+
+    async setFieldScript(script: string) {
+      log.debug('set script = ' + script);
+      await (
+        await (await testSubjects.find('valueRow')).findAllByCssSelector(
+          '[data-test-subj="toggle"]'
+        )
+      )[0].click();
+      await browser.pressKeys(browser.keys.TAB);
+      await new Promise((resolve) => setTimeout(resolve, 1000 * 5));
```

Comment:
This still needs to be addressed. Do we have functionality that allows us to work with the monaco editor?

Reply:
Not that I know of. I wonder if the monaco editor has some sort of `onFocus` or `onMount` that we could leverage to add a `data-` attribute on the wrapper. Then this test could wait for the attribute to be in the DOM.
Thougts on this @alisonelizabeth ?

Reply:
>Do we have functionality that allows us to work with the monaco editor?

I'm also not aware of any as well. Current solution may become flaky issue, but  [looks](https://stackoverflow.com/questions/41040985/unable-to-get-the-text-inside-the-monaco-editor-using-protractor) like it is not that easy to test monaco editor in general.

To get much out of the test and don't worry about stability, I would address how to:
- locate input element directly
- check its state
- verify its value

Even if we can't do it via native web driver API, but can with pure js script, I think it is beneficial.

Reply:
@dmlemeshko I was able to locate the textarea with js. What do you think of my current code?

Reply:
@mattkime, FYI, we have here a tiny bit with monaco: https://github.com/elastic/kibana/blob/68ed17a486e4059fd3c903df7cb0a404f9d4872d/x-pack/test/functional/services/dashboard/drilldowns_manage.ts#L132-L137

Reply:
> Not that I know of. I wonder if the monaco editor has some sort of onFocus or onMount that we could leverage to add a data- attribute on the wrapper. 

The `CodeEditor` component does expose an `editorDidMount` prop, which is emitted when the editor is mounted and provides access to the [monaco `editor` instance](https://microsoft.github.io/monaco-editor/api/interfaces/monaco.editor.icodeeditor.html).